volume-providers/softlayer/utils/storage_utils.go

```diff
@@ -542,3 +542,35 @@ func ConvertToLocalSnapshotObject(storageSnapshot datatypes.Network_Storage, log
 
 	return snapshot
 }
+
+//GetSubnetListFromIDs returns list of Network_Subnet from subnetIDs
+func GetSubnetListFromIDs(logger *zap.Logger, session backend.Session, subnetIDs []string) ([]datatypes.Network_Subnet, error) {
+	subnetList := []datatypes.Network_Subnet{}
+	subnetService := session.GetNetworkSubnetService()
+	for _, subnetID := range subnetIDs {
+		subnetIDInt, _ := strconv.Atoi(subnetID)
+		subnet, err := subnetService.ID(subnetIDInt).GetObject()
+		if err == nil {
+			subnetList = append(subnetList, subnet)
+		} else {
+			logger.Error("Unable to get the subnet by id ", zap.Error(err), zap.String("subnetId", subnetID))
+			return nil, err
```

Comment:
Do you mean to say if we are fail to get subnet for any subnetID then we should return nil i.e fail to get for all,  please check in storage plugin what we are doing

Reply:
done

Reply:
All-or-None is the common strategy we should follow for any transaction . In can of partial  processing, user will have to keep track on what is failed in what is applied